Source/Applications/SystemCenterNotification/Controllers/EmailTypeController.cs

```diff
@@ -240,9 +226,13 @@ void ConfigureRequest(HttpRequestMessage request)
                     .GetAwaiter()
                     .GetResult();
 
-                IEnumerable<DataSourceResponse> results = JsonConvert
-                    .DeserializeObject<IEnumerable<DataSourceResponse>>(responseJSON)
-                    .Select(r => new DataSourceResponseTSX(r));
+                JArray results = JArray.Parse(responseJSON);
+
+                for(int index = 0; index < results.Count(); index++)
+                {
+                    XElement data = JsonConvert.DeserializeObject<XElement>(results[index]["Data"].ToString());
```

Comment:
Is it necessary to deserialize it if you're converting it right back into a string?

Reply:
Yes, we its not in the right format without that unless we just want to send it like that in the first place from XDA, but I'm not sure if this endpoint is used anywhere else. This is essentially what the code was doing before, just removed the custom class that it was squirreled away in.